relay-server/src/services/health_check.rs

```diff
@@ -76,39 +183,34 @@ impl HealthCheckService {
             });
         }
 
-        match message {
-            IsHealthy::Liveness => true,
-            IsHealthy::Readiness => {
-                if self.is_shutting_down.load(Ordering::Relaxed) {
-                    return false;
-                }
+        if matches!(message, IsHealthy::Liveness) {
+            return Status::Healthy;
+        }
 
-                if self.config.requires_auth()
-                    && !upstream.send(IsAuthenticated).await.unwrap_or(false)
-                {
-                    return false;
-                }
+        if self.is_shutting_down.load(Ordering::Relaxed) {
+            return Status::Unhealthy;
+        }
 
-                if !self
-                    .aggregator
-                    .send(AcceptsMetrics)
-                    .await
-                    .unwrap_or_default()
-                {
-                    return false;
-                }
+        let (sys_mem, auth, agg, proj) = tokio::join!(
```

Comment:
nit: I wonder if we could use [`tokio::try_join`](https://docs.rs/tokio/latest/tokio/macro.try_join.html) to not wait on the futures after one of them resolves to unhealthy.

Reply:
Yeah, I thought about this too but didn't do it because of the awkward signature, would probably have to return `Result<(), ()>` from every probe.

Do you think I should change it?

Reply:
Nah, I don't think it's worth the effort (that's why nit), I'd just ship it. If we bikeshed, a `Result` could make sense from a semantical point of view (a failed health check could be an error?), plus allow the `try_join`.